zanata-war/src/main/java/org/zanata/action/ProjectHome.java

```diff
@@ -122,11 +134,82 @@
     @In
     private CopyTransOptionsModel copyTransOptionsModel;
 
+    // This property is present to keep the filter in place when the region with
+    // the filter box is refreshed.
+    @Getter
+    @Setter
+    private String enabledLocalesFilter = "";
+
+    @Getter
+    @Setter
+    private String disabledLocalesFilter;
+
+    /**
+     * A separate map is used, rather than binding the alias map from the
+     * project directly. This is done so that empty values are not added to the
+     * map in every form submission, and so that a value entered in the field
+     * for a row is not automatically updated when a different row is submitted.
+     */
+    @Getter
+    @Setter
+    private Map<LocaleId, String> enteredLocaleAliases = Maps.newHashMap();
+
+    @Getter
+    @Setter
+    private Map<LocaleId, Boolean> selectedEnabledLocales = Maps.newHashMap();
+
+    // Not sure if this is necessary, seems to work ok on selected disabled
+    // locales without this.
+    public Map<LocaleId, Boolean> getSelectedEnabledLocales() {
+        if (selectedEnabledLocales == null) {
```

Comment:
@aeng I actually think I need to use the opposite approach - I should set `selectedEnabledLocales` to `null` when the list of enabled locales changes, otherwise it will not contain the correct set of locales. I will sort that out.